### PR TITLE
Add PortForward to runtime API.

### PR DESCRIPTION
**Pull request by vmarmol**
/cc @dchen1107 @yifan-gu

### CODE REVIEW

File: pkg/kubelet/container/runtime.go [hunk 1]
```diff
@@ -66,6 +66,8 @@ type Runtime interface {
 	// Attaches the processes stdin, stdout, and stderr. Optionally uses a
 	// tty.
 	ExecInContainer(container api.Container, pod *api.Pod, stdin io.Reader, stdout, stderr io.WriteCloser, tty bool)
+	// Forward the specified port from the specified pod to the stream.
+	PortForward(pod *api.Pod, port uint16, stream io.ReadWriteCloser)
```
Comment (yifan-gu) [inline pkg/kubelet/container/runtime.go]: I found the current implementation takes a `*kubecontainer.Pod`. But that's the identical. Both work :)

Comment (vmarmol) [inline pkg/kubelet/container/runtime.go]: That's a good point. We need to standardize on one or the other. Right now everything in the interface takes api.Pod even when it doesn't need to. We can start with this one.

Changed it to Pod.

**Comment by dchen1107** [comment]
LGTM. Are we going to implement PortForward to rkt support now?

File: pkg/kubelet/container/runtime.go (hunk 1 above)
Comment (yifan-gu) [inline pkg/kubelet/container/runtime.go]: SGTM.

**Comment by yifan-gu** [comment]
@dchen1107 I think I can implement that. It's almost the same as the docker one. We just need the pid for the "pod", which is available via `rkt status`

**Comment by dchen1107** [comment]
SGTM. Will merge it once it is green.

**Comment by vmarmol** [comment]
Travis is green! (finally...)

**Pull request merged**